### PR TITLE
Chris/improve release docs

### PR DESCRIPTION
**Pull request by Xophmeister**
Edited and amended the release instructions to make them easier to follow, with examples where necessary. Also removed the Google CLA boilerplate, as it doesn't seem relevant.

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -1,71 +1,137 @@
...
+- None
+
+<!-- Links -->
+[start]: ./start
+[versions]: ./haskell/private/versions.bzl
+[readme]: ./README.md
+[changelog]: ./CHANGELOG.md
+[usecases]: ./docs/haskell-use-cases.rst
+[releases]: https://github.com/tweag/rules_haskell/releases
+[checklist]: https://github.com/tweag/rules_haskell/blob/52f867daa2c3e67e1afff9a0460666ef9083e714/MAINTAINERS.md?plain=1#L29-L86
```
Comment (aherrmann) [inline MAINTAINERS.md:137]: Isn't this likely to go out of sync on future changes to this doc? Maybe it's good without the link to itself.

File: MAINTAINERS.md [hunk 2]
```diff
@@ -1,71 +1,137 @@
...
+      feature).
+  - [ ] Check the list [below](#remove-these-prs-from-minor-releases)
+        for PRs that have been explicitly marked for removal, if any.
+- [ ] Amend the [`CHANGELOG`][changelog] by summarising all significant
+      pull requests since the last release (see
+      [below](#generating-the-pr-list-for-the-changelog)). Specifically:
+  - [ ] Add a "Highlights" section for major improvements/changes.
+  - [ ] Create "Added", "Removed", "Changed" and "Fixed" sections, as
+        necessary.
+  - [ ] If relevant, add corresponding PRs to the changes.
```
Comment (aherrmann) [inline MAINTAINERS.md:49]:
```suggestion
  - [ ] If relevant, add links to the corresponding PRs to the entries.
```
Perhaps a bit clearer?

**Review by aherrmann — APPROVED** [review]
Thank you, that looks much better!

Only two small comments.

File: MAINTAINERS.md (hunk 1 above)
Comment (Xophmeister) [reply MAINTAINERS.md:137]: It is, but I figured that this list would be relatively static beyond overhauls like this one. Still, I take your point that such changes can and will happen in the future and updating the link will likely be forgotten about.

**Pull request merged**